etc/rsync.profile

```diff
@@ -0,0 +1,54 @@
+# Firejail profile for rsync
+# Description: a fast, versatile, remote (and local) file-copying tool
+# This file is overwritten after every install/update
+quiet
+# Persistent local customizations
+include rsync.local
+# Persistent global definitions
+include globals.local
+
+# This profile is focused on using rsync as an client for downlaoding, it is not writen for using rsync
+#  - as an daemon (rsync --daemon)
+#  - creating backup of $HOME or a full system
+
+blacklist /tmp/.X11-unix
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+# Uncomment or add to rsync.local to enable extra hardening
+#whitelist ${DOWNLOADS}
```

Comment:
TBH I'm not sure why only Downloads would need to be whitelisted... but then I pretty much only use rsync for backing up my /home so perhaps I'm just not being imaginative enough :smile:

Reply:
The profile should be used for download (as mentioned above).  If you wan't to do backup and we include disable-*.inc there is an issue.

Reply:
@rusty-snake could you elaborate on the issue when doing backups?

Reply:
If you want backups from ...
 * various configurations and `include disable-programms.inc` is there
 * your documents and `include disable-xdg.inc` is there
 * your firejail profile customisations, even with `--noprofile`
...
=> If you want backups of your full ${HOME} and not only a few files you may want to omit firejail.

Reply:
@rusty-snake IMO it's probably not a great idea to include the backup-breaking options then. I don't know how common it is to use rsync directly for backups (I *assumed* it was common just because that's the way I use it!).

However, Linux Mint by default (if I understand it correctly) heavily encourages using TimeShift for backups before updates -- I believe their update manager may create the backups automatically (via timeshift) but I could be wrong there. Anyhow that's one major distro, at least, that uses timeshift heavily. For everyone not on BTRFS, then timeshift uses rsync for the backup.

TimeShift normally wouldn't include $HOME in the backup but it's easily configured to do so. Any Mint user who uses timeshift (many/most?) without btrfs (many/most?) and who also backup their $HOME with timeshift (I expect a good-sized minority) will suddenly have incomplete or failing backups.

TL,DR: If we commit an rsync profile that makes backups harder, I expect we'll see a lot of unhappy Mint users. :wink: Personally I think we shouldn't do this.